script.js

```diff
@@ -0,0 +1,68 @@
+const johnTeamScores = [89, 20, 103];
+const mikeTeamScores = [116, 94, 123];
+const johnsAvrg = calcAverage(johnTeamScores);
+const mikesAvrg = calcAverage(mikeTeamScores);
+const sameAvrg = ' have the same score';
+
+const higherAverage = {
+    winner: 'John and Mike',
+    highestAvrg: '\'s team has the highest average score of ',
+    score: '',
+
+    setWinner() {
+
+        if (johnsAvrg !== mikesAvrg && johnsAvrg > mikesAvrg) {
+            this.winner = 'John';
+            this.score = johnsAvrg;
+        }
+        else if (johnsAvrg !== mikesAvrg) {
+            this.winner = 'Mike';
+            this.score = mikesAvrg;
+        }
+        else {
+            this.highestAvrg = sameAvrg;
+        }
+    }
+};
+
+higherAverage.setWinner();
+
+function calcAverage(scores) {
+    
+    let sum = 0;
+
+    scores.forEach(score => {
+        sum += score;
+    });
+
+    return sum / scores.length;
+}
+
+//Extra
+
+const maryTeamScores = [97, 134, 105];
+const maryAvrg = calcAverage(maryTeamScores);
+
+if (maryAvrg > johnsAvrg && maryAvrg > mikesAvrg) {
+    higherAverage.winner = 'Mary';
+    higherAverage.score = maryAvrg;
+}
+
+else if (maryAvrg > johnsAvrg && maryAvrg == mikesAvrg) {
+    higherAverage.winner = 'Mary and Mike';
+    higherAverage.highestAvrg = sameAvrg;
+}
+
+else if (maryAvrg > mikesAvrg && maryAvrg == johnsAvrg) {
+    higherAverage.winner = 'Mary and John';
+    higherAverage.highestAvrg = sameAvrg;
+}
+
+else if (maryAvrg === mikesAvrg && maryAvrg === johnsAvrg) {
+    higherAverage.winner = 'The three teams are tied';
+    higherAverage.highestAvrg = '';
+}
+
+const avrg = higherAverage.winner + higherAverage.highestAvrg + higherAverage.score;
```

Comment:
On Es6 were introduce the [template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals). We can use it here instead of storing the complete string value; just need to put it twice in case there was a tie.

Reply:
Like this?